File: 2427-first-letter-to-appear-twice/2427-first-letter-to-appear-twice.java
class Solution {
    public char repeatedCharacter(String s) {
        Set<Character>seen=new HashSet<>();
        for(char ch:s.toCharArray()){
            if(seen.contains(ch)){
                return ch;
            }
            else{
                seen.add(ch);
            }
        }
        return '\0';
    }
}

File: 2427-first-letter-to-appear-twice/2427-first-letter-to-appear-twice.py
class Solution:
    def repeatedCharacter(self, s: str) -> str:
        seen = set()  # Use a set to store characters we've seen so far
        for char in s:
            if char in seen:  # If the character is already in the set, it's repeated
                return char
            else:
                seen.add(char)  # Add the character to the set
        return ''  # This line is redundant for the problem but included for completeness
